rspace/src/main/scala/coop/rchain/rspace/history/RadixTree.scala

```diff
@@ -0,0 +1,987 @@
+package coop.rchain.rspace.history
+
+import cats.Parallel
+import cats.effect.Sync
+import cats.syntax.all._
+import coop.rchain.shared.syntax._
+import coop.rchain.store.KeyValueTypedStore
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+import scala.collection.concurrent.TrieMap
+
+object RadixTree {
+
+  /**
+    * Described items which contain in [[Node]].
+    */
+  sealed trait Item
+
+  /**
+    * Empty item
+    */
+  final case object EmptyItem extends Item
+
+  /**
+    * Item which contain data.
+    */
+  final case class Leaf(prefix: ByteVector, value: ByteVector) extends Item
+
+  /**
+    * Item which contain pointer for child [[Node]].
+    */
+  final case class NodePtr(prefix: ByteVector, ptr: ByteVector) extends Item
+
+  /**
+    * Base type for nodes in Radix History.
+    *
+    * Must contain 256 [[Item]]s.
+    */
+  type Node = Vector[Item]
+
+  /**
+    * Number of items always constant.
+    */
+  val numItems = 256
+
+  /**
+    * Empty node consists only of [[EmptyItem]]s.
+    */
+  val emptyNode: Node = (0 until numItems).map(_ => EmptyItem).toVector
+
+  /**
+    * Binary codecs for serializing/deserializing Node in Radix tree
+    *
+    * {{{
+    * Coding structure for items:
+    *   EmptyItem                   - Empty (not encode)
+
+    *   Leaf(prefix,value)    -> [item number] [second byte] [prefix0]..[prefixM] [value0]..[value31]
```

Comment:
I finally figure out `item number` is the index of the `item` instead of the identical byte for enum `item`.

Could you please change the term `item number` into `item index`?

Reply:
Do I understand correctly?

Reply:
I think `index` is better than `number`. What do you think?

Reply:
You are right. Thanks!